pubsub/google/cloud/pubsub_v1/subscriber/_helper_threads.py

```diff
@@ -84,6 +97,16 @@ def stop(self, name):
         if helper_thread is None:
             return
 
+        if helper_thread.thread is _current_thread():
+            # The current thread cannot ``join()`` itself but it can
+            # still send a signal to stop.
+            _LOGGER.debug('Cannot stop current thread %s', name)
+            helper_thread.queue.put(STOP)
+            # We return and stop short of ``pop()``-ing so that the
+            # thread that invoked the current helper can properly stop
+            # it.
+            return
```

Comment:
So this happens when `ConsumeBidirectionalStream` worker (which has `_blocking_consume` as its target) encounters a non-retry-able exception.

This is essentially fine, but will leave the worker in a "zombie" state until `Policy.close()` is called.

I originally considered adding `self._helper_threads.pop(name, None)` in this branch (i.e. do **everything except** the `join()`) but I realized it'd make more sense for a caller in the **correct thread** to be able to safely call `join()` and un-zombie the thread.